agent_api_http/src/v1/identity/keys/mod.rs

```diff
@@ -0,0 +1,165 @@
+    context.key_removal_saga.remove_key(managed_key_id).await;
+    Ok(StatusCode::NO_CONTENT)
+}
+
+#[derive(Deserialize)]
+pub struct PostRenameAlias {
+    pub key_id: String,
+    pub new_alias: String,
+}
+
+pub(crate) async fn rename_key_alias(
+    State(context): State<IdentityContext>,
+    Json(payload): Json<PostRenameAlias>,
+) -> Result<StatusCode, ApiError> {
+    let managed_key_id = query_handler(&payload.key_id, &context.state.query.managed_key)
+        .await
+        .map_err(|e| {
+            tracing::error!("Query failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?
+        .map(|key_view| key_view.managed_key_id)
+        .ok_or_else(|| ApiError::new(StatusCode::NOT_FOUND))?;
+
+    let command = ManagedKeyCommand::UpdateKeyAlias {
+        new_alias: payload.new_alias,
+    };
+
+    command_handler(&managed_key_id, &context.state.command.managed_key, command)
+        .await
+        .map_err(|e| {
+            tracing::error!("Saga failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?;
+    Ok(StatusCode::NO_CONTENT)
+}
+
+#[derive(Deserialize)]
+pub struct PostSetSigningKey {
+    pub key_id: String,
+}
+
+pub(crate) async fn set_signing_key(
+    State(context): State<IdentityContext>,
+    Json(payload): Json<PostSetSigningKey>,
+) -> Result<StatusCode, ApiError> {
+    let managed_key_id = query_handler(&payload.key_id, &context.state.query.managed_key)
+        .await
+        .map_err(|e| {
+            tracing::error!("Query failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?
+        .map(|key_view| key_view.managed_key_id)
+        .ok_or_else(|| ApiError::new(StatusCode::NOT_FOUND))?;
+
+    let command = ManagedKeyCommand::SetSigningKey {};
+
+    command_handler(&managed_key_id, &context.state.command.managed_key, command)
+        .await
+        .map_err(|e| {
+            tracing::error!("Saga failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?;
+    Ok(StatusCode::NO_CONTENT)
+}
+
+#[axum_macros::debug_handler]
+pub(crate) async fn list_all(
+    State(context): State<IdentityContext>,
+) -> Result<(StatusCode, Json<Vec<ManagedKeyDto>>), ApiError> {
+    let view = query_handler("all_managed_keys", &context.state.query.all_managed_keys)
+        .await
+        .map_err(|e| {
+            tracing::error!("Query failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?;
+
+    // AllManagedKeysView contains a HashMap of ManagedKeyViews
+    let keys = view
+        .unwrap()
+        .managed_keys
+        .into_values() // Iterate over the 'ManagedKeyView' items inside
+        .map(|key_view| ManagedKeyDto {
+            managed_key_id: key_view.managed_key_id,
+            key_id: key_view.key_id,
+            alias: key_view.alias,
+            signing_algorithm: key_view.signing_algorithm,
+        })
```

Comment:
This endpoint should filter out all the views which `is_removed` field is set to `false`, e.g:
```rust
     .filter_map(|key_view| {
            (!key_view.is_removed).then(|| ManagedKeyDto {
                managed_key_id: key_view.managed_key_id,
                key_id: key_view.key_id,
                alias: key_view.alias,
                signing_algorithm: key_view.signing_algorithm,
            })
        })
```